fix(connect envoy): set initial_fetch_timeout to wait for initial xDS… (#17317)
hashicorp/consul: .changelog/17317.txt

```release-note:bug

> connect: fix bug with Envoy starting with partial configuration by wait indefinitely for complete, initial xDS configuration.

Reviewer: ```suggestion
connect: fix a bug with Envoy potentially starting with incomplete configuration by not waiting enough for initial xDS configuration.
```
